cdn/publicfeeds-cdn.yml

```diff
@@ -70,30 +70,54 @@ Resources:
           exports.handler = (event, context, callback) => {
             const request = event.Records[0].cf.request;
 
-            if (request.uri.startsWith('/f/3954/feed-rss.xml')) {
-              callback(null, { status: 301, headers: { location: [{ key: 'Location', value: 'https://feeds.megaphone.fm/technically-optimistic' }] } });
-            } else if (request.uri.startsWith('/f/')) {
+            // The `/f` path is reserved to indicate an f.prxu.org feed origin. The `/f`
+            // path part must be removed to make a valid origin request.
+            if (request.uri.startsWith('/f/')) {
               request.uri = request.uri.slice(2);
+
+              const headUrl = `https://${request.origin.custom.domainName}${request.uri}`;
+
+              // Check the feed file for a redirect marker
+              fetch(headUrl, { method: 'HEAD' }).then((res) => {
+                if (res.headers.has('x-amz-meta-dovetail-feed-cdn-redirect')) {
+                  // The feed file in S3 being requested has been marked with a
+                  // redirect header. Respond with a 301 to the given URL, rather than
+                  // with the actual feed.
+                  callback(null, {
+                    status: 301,
+                    headers: {
+                      location: [
```

Comment:
Does this need a `Cache-Control` header on it?  Not for the client really (301s are permanent)... but for the CDN?  I'm not sure either publicfeeds or f.prxu.org have a default TTL.  So they may just cache it for a real-long-time if you don't provide a max age.